packages/SwingSet/test/test-xsnap-errors.js

```diff
@@ -0,0 +1,95 @@
+/* global require, __dirname, process */
+// @ts-check
+// eslint-disable-next-line import/order
+import { test } from '../tools/prepare-test-env-ava.js';
+import path from 'path';
+import { spawn } from 'child_process';
+import bundleSource from '@agoric/bundle-source';
+
+import { makeXsSubprocessFactory } from '../src/kernel/vatManager/manager-subprocess-xsnap.js';
+import { makeStartXSnap } from '../src/controller.js';
+import { capargs } from './util.js';
+
+function suppressUnhandled(t, message) {
+  process.on('unhandledRejection', (reason, promise) => {
```

Comment:
I'm not super familiar with ava but my experience with other test frameworks is that this is leaky.

The test should as much as possible manually cleanup these global event handlers after the test has run so that events in future tests don't interact weirdly.

Reply:
I stipulate that it should, but in ava, each test script runs its own node process, so I don't think this is a correctness issue.